app/questionnaire/router.py

```diff
@@ -346,13 +345,16 @@ def _is_section_enabled(self, section: Mapping) -> bool:
         )
 
     @staticmethod
-    def get_next_block_url(location: Location, routing_path: RoutingPath) -> str:
+    def get_next_block_url(
+        location: Location, routing_path: RoutingPath, **kwargs: str
+    ) -> str:
```

Comment:
We also need to support the persistence of the query param for `get_previous_location_url`. At the moment, if you open up say 3 new questions, if you click `Previous` the query string is lost and when you subsequently go forward and complete the new questions, we won't jump back to the summary. 

Have confirmed with Donna that even if a user clicks `Previous`, soon as the section is re-complete, we should jump back to where we came from. We already have `return_to` in scope for `get_previous_location_url`, it is just not persisted.

Reply:
True, shouldn't be difficult